src/components/Planning/PlanningTable/PlanningRow/ProjectRow/ProjectCell.tsx

```diff
@@ -30,6 +39,104 @@ const addActiveClassToProjectRow = (projectId: string) => {
   document.getElementById(`project-row-${projectId}`)?.classList.add('active');
 };
 
+const getCellTypeUpdateRequestData = (cell: IProjectCell, phase: string): IProjectRequest => {
+  const { type, startYear } = cell;
+  const req: IProjectRequest = { finances: { year: startYear } };
+  /**
+   * Returns the first/last valid construction/planning cell and sets any inbetween cell finances to null.
+   * Traverses backwards to find a planning cell and forward to find a construction cell.
+   * @param cellType the type of cell to find
+   * @returns IProjectCell
+   */
+  const traverseAndSetGapCellFinancesNull = (cellType: 'construction' | 'planning') => {
+    let head: IProjectCell | null = cellType === 'construction' ? cell.next : cell.prev;
+    while (head && !head.type.toLowerCase().includes(cellType)) {
+      // Have to use a if condition to check as other operators are still raising typescript errors
+      if (req.finances) {
+        (req.finances[head.financeKey as keyof IProjectFinancesRequestObject] as null) = null;
+      }
+      head = cellType === 'construction' ? head.next : head.prev;
+    }
+    return head;
+  };
+  /**
+   * Traverses backwards from current cell to get the first valid planning cell
+   * @returns IProjectCell
+   */
+  const getFirstPlanCellBehind = () => {
+    return traverseAndSetGapCellFinancesNull('planning');
+  };
+  /**
+   * Traverses forward from current cell to get the first valid construction cell
+   * @returns IProjectCell
+   */
+  const getFirstConCellAhead = () => {
+    return traverseAndSetGapCellFinancesNull('construction');
+  };
+  /**
+   * Gets EndOfYear date from the first valid planning cell behind
+   * @returns string
+   */
+  const getDateFromFirstPlanCellBehind = () => {
+    const planCellBehind = getFirstPlanCellBehind();
+    return planCellBehind ? createDateToEndOfYear(planCellBehind.year) : null;
+  };
+  /**
+   * Gets StartOfYear date from the first valid construction cell ahead
+   * @returns string
+   */
+  const getDateFromFirstConCellAhead = () => {
+    const conCellAhead = getFirstConCellAhead();
+    return conCellAhead ? createDateToStartOfYear(conCellAhead.year) : null;
+  };
+  switch (type) {
+    case 'planningEnd':
+      if (cell.prev?.type.includes('planning')) {
```

Comment:
Move the content of each switch case to update functions to make it more readable.

  switch (type) {
    case 'planningEnd':
      updatePlanningEnd();
    case 'constructionStart':
      updateConstructionStart();
    ...
 }